docs/src/main/asciidoc/application-configuration-guide.adoc

```diff
@@ -360,6 +360,48 @@ You can override these runtime properties with the following mechanisms (in decr
 
 NOTE: Environment variables names are following the conversion rules of link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#default-configsources[Eclipse MicroProfile]
 
+== Using local configuration file during Development
+As an alternative to defining some configuration variables via host environment variables during development, you can define configuration variables in a local configuration file via a `quarkus.local-config` property.
+This is practical as it allows each member of your development team to customize and manage the development environment via a file.
+Doing so, Quarkus will load configuration variables having higher priority than variable defined in a `application.properties` file but with a lower priority than host environment variables.
+
+See the example below.
+
+Let's create a `src/main/resources/.local-config.properties` file and add the following variable.
+
+[source,shell]
+--
+GREETING_MESSAGE=hello world
+--
+
+Edit the main configuration `src/main/resources/application.properties` file so that it looks like:
+
+[source,properties]
+----
+greeting.message = ${GREETING_MESSAGE}
```

Comment:
This example is a little weird because `GREETING_MESSAGE` is the environment-style name of the `greeting.message` property.   So it's really unclear from the example whether the `GREETING_MESSAGE` in the `.local-config.properties` is meant to (or will) override the value or to provide the resolution for the expression.

Reply:
I should change the variable name a bit.